runner/engine_test.go

```diff
@@ -1058,3 +1058,56 @@ include_file = ["main.sh"]
 	}
 	assert.Equal(t, []byte("modified"), bytes)
 }
+
+type testExiter struct {
+	t          *testing.T
+	expectCode int
+}
+
+func (te testExiter) Exit(code int) {
+	if code != te.expectCode {
+		te.t.Fatalf("expected exit code %d, got %d", te.expectCode, code)
+	}
+}
```

Comment:
This is OK, but I would add a boolean to validate it was called

```suggestion
type testExiter struct {
	t          *testing.T
	called bool
	expectCode int
}

func (te testExiter) Exit(code int) {
	te.called = true
	if code != te.expectCode {
		te.t.Fatalf("expected exit code %d, got %d", te.expectCode, code)
	}
}
```

Reply:
done